torch/nn/utils/_expanded_weights/group_norm_expanded_weights.py

```diff
@@ -0,0 +1,58 @@
+from functools import reduce
+import operator
+import torch
+import torch.nn.functional as F
+from .expanded_weights_impl import implements_per_sample_grads
+from .expanded_weights_utils import standard_kwargs, \
+    forward_helper, set_grad_sample_if_exists, unpack_expanded_weight_or_tensor
+from typing import List, Optional
+
+@implements_per_sample_grads(F.group_norm)
+class GroupNormPerSampleGrad(torch.autograd.Function):
+    @staticmethod
+    def forward(ctx, kwarg_names, _, *expanded_args_and_kwargs):
+        expanded_args, expanded_kwargs = standard_kwargs(kwarg_names, expanded_args_and_kwargs)
+        input, num_groups = expanded_args
+        N = input.shape[0]
+        C = input.shape[1]
+        HxW = reduce(operator.mul, input.shape[2:], 1)
+        weight, bias, eps = expanded_kwargs['weight'], expanded_kwargs['bias'], expanded_kwargs['eps']
+        output, mean, rstd = forward_helper(torch.native_group_norm, (input, weight, bias, N, C, HxW, num_groups, eps), {})
+        ctx.input, ctx.num_groups = input, num_groups
+        ctx.weight, ctx.bias, ctx.eps = weight, bias, eps
```

Comment:
Optimization: we don't have to save weight, bias if none of them require gradients, right? (this is the case where someone defines nn.GroupNorm with affine=False.

Doesn't need to be handled in this PR, could be a followup. Also because I am not sure how we'd go about testing this (is there a test case for nn.GroupNorm with affine=False?)

Reply:
Added optimization (weight will still be needed for group norm backward, but able to optimize it out if neither input nor weight require gradients)! Group Norm does have an affine=False test but might want to double check when input + weight don't require grad but bias does...